### PR TITLE
Use binary readiness probe for Elasticsearch instead of bash script

### PR DESCRIPTION
**Pull request by Brian-McM**
The native bash script readiness probe that comes with Elasticsearch uses curl which we've removed because of CVEs, so we've rewritten the readiness probe in golang, compiled it into a binary, and added it to the Elasticsearch image.

## For PR author

- [x] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [x] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Review by matthewdupre — APPROVED** [review]

**Comment by matthewdupre** [comment]
CI failure looks unrelated at a glance

**Pull request merged**
